store/store.go

```diff
@@ -1375,11 +1376,21 @@ func (s *Store) Download(ctx context.Context, name string, targetPath string, do
 		return err
 	}
 	defer func() {
+		fi, _ := w.Stat()
 		if cerr := w.Close(); cerr != nil && err == nil {
 			err = cerr
 		}
 		if err != nil {
-			os.Remove(w.Name())
+			doRemove := dlOpts == nil || !dlOpts.LeavePartialOnError
+			if !doRemove {
```

Comment:
at the sake of being nitpicky, this would read a little bit easier to me if it was:

```go
			if !doRemove && fi != nil && fi.Size() == 0 {
				// also do remove if the file is empty
				doRemove = true
			}
```

it's just kind of confusing to initially see

```go
if !thing {

}
if thing {

}
```

Reply:
Yeah, I struggled a bit with that one.
I could also write it as
```go
		if err != nil && (dlOpts == nil || !dlOpts.LeavePartialOnError || (fi != nil && fi.Size() == 0)) {
			os.Remove(w.Name())
		}
```
wdyt?

Reply:
IMHO the `()` is a bit more confusing there to see which conditions combine with which

Reply:
that looks best actually, no nested paranthesis and only `||` between the checks